sr_port/stp_expand_array.c

```diff
@@ -15,18 +18,18 @@
 #include "stp_parms.h"
 #include "longcpy.h"
 
-GBLREF mstr **stp_array;
-GBLREF int stp_array_size;
+GBLREF mstr		**stp_array;
+GBLREF gtm_uint64_t	stp_array_size;
 
 void stp_expand_array(void)
 {
-	mstr **a;
-	int n;
+	mstr		**a;
+	gtm_uint64_t	n;
 
 	n = stp_array_size;
-	stp_array_size += STP_MAXITEMS;
+	stp_array_size = ((MAXINT4 > n) ? (n * 2) : (n + MAXINT4));
 	a = stp_array;
-	stp_array = (mstr **) malloc(stp_array_size * SIZEOF(mstr *));
+	stp_array = (mstr **)malloc(stp_array_size * SIZEOF(mstr *));
 	longcpy((uchar_ptr_t)stp_array, (uchar_ptr_t)a, n * SIZEOF(mstr *));
```

Comment:
Opportunity for longcopy->memcpy (optional)

Reply:
Done here and all other remaining usages.